ddcommon-ffi/src/tracer_metadata.rs

```diff
@@ -0,0 +1,85 @@
+// Copyright 2023-Present Datadog, Inc. https://www.datadoghq.com/
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::slice::CharSlice;
+#[cfg(target_os = "linux")]
+use ddcommon::tracer_metadata::AnonymousFileHandle;
+use ddcommon::tracer_metadata::{self, TracerMetadata};
+use std::os::raw::c_int;
+
+/// C-compatible representation of an anonymous file handle
+#[repr(C)]
+pub struct TracerMemfdHandle {
+    /// File descriptor (relevant only on Linux)
+    pub fd: c_int,
+    /// Whether the handle is valid
+    pub is_valid: bool,
```

Comment:
Do we actually need that, or can we just return `c_int` and check for -1 in code?

Reply:
Good point, I removed it, instead I'm now wrapping [the result here](https://github.com/DataDog/libdatadog/pull/985/commits/df8d0ae1b9cf4d7b69a72b3bb5ad54d8174ecc2f#diff-37c0cf3172b52d455f9826a760faa521726c026954021d9763ec85958c33a0f2) to get the error message on the tracer side and be able to log it, let me know wyt thanks

Reply:
Sounds good :-)